framework/util/platform.h

```diff
@@ -455,8 +490,51 @@ inline int32_t MakeDirectory(const char* filename)
     return mkdir(filename, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
 }
 
+inline size_t GetSystemPageSize()
+{
+    return getpagesize();
+}
+
+// Memory allocation without extra space for private info like with new or malloc
+inline void* AllocateRawMemory(size_t aligned_size, bool use_write_watch = false)
+{
+    assert(aligned_size > 0);
+
+    if (aligned_size > 0)
+    {
+        void* memory = mmap(nullptr, aligned_size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+
+        if (memory == MAP_FAILED)
+        {
```

Comment:
`GFXRECON_LOG_ERROR()` To get a message output before the segfault in the caller?

Reply:
I thought it's a caller responsibility now
But maybe better log here, but it's harder to inject logging because the logging.h depends on platform.h
Will try to handle this

Reply:
Okay, just a thought. maybe at the call sites then. ... even though if this fails, things are going very badly.

Reply:
Yep, gdb will point to a failed line

Added extra check to the client side in replay_consumer